### PR TITLE
Immutable checkpoint api update

### PR DESCRIPTION
**Pull request by Doy-lee**
* Remove unused result variable
* Return the checkpoint to make it more useful
* Copy and use the checkpoint handling logic from checkpoints.cpp

@jagerman

### CODE REVIEW

File: src/blockchain_db/lmdb/db_lmdb.cpp [hunk 1]
```diff
@@ -3966,14 +3966,43 @@ std::vector<checkpoint_t> BlockchainLMDB::get_checkpoints_range(uint64_t start,
...
+      // so we can't reorg past that height. If it's predefined, that's ok as
+      // well, we can't reorg past that height so irrespective, always accept
+      // the height of this next checkpoint.
+      immutable_checkpoint_ptr = &checkpoints[1];
+    }
+  }
+  else
+  {
+    immutable_checkpoint_ptr = &checkpoints[0];
+  }
```
Comment (jagerman) [inline src/blockchain_db/lmdb/db_lmdb.cpp]: Slightly simpler logic (but this is subjective):  (The comments are great; just omitting them here for brevity).
```C++
checkpoint_t const *immutable_checkpoint_ptr = nullptr;
if (checkpoints[0].type != checkpoint_type::service_node)
    immutable_checkpoint_ptr = &checkpoints[0];
else if (checkpoints.size() > 1)
    immutable_checkpoint_ptr = &checkpoints[1];
else
    return false;

if (immutable_checkpoint) *immutable_checkpoint = *immutable_checkpoint_ptr;
```

Comment (jagerman) [reply src/blockchain_db/lmdb/db_lmdb.cpp]: Another simplification: the pointer into the vector seems like unnecessary indirection versus just storing the value directly: (again omitting comments for brevity).

```C++
checkpoint_t immutable;
if (checkpoints[0].type != checkpoint_type::service_node)
    immutable = checkpoints[0];
else if (checkpoints.size() > 1)
    immutable = checkpoints[1];
else
    return false;

if (immutable_checkpoint) *immutable_checkpoint = immutable;

Comment (Doy-lee) [reply src/blockchain_db/lmdb/db_lmdb.cpp]: Well I mean the non-ptr way is a copy right? You presuming this is optimised away in release? But also realised we can std::move to steal the signatures here as well.

Comment (jagerman) [reply src/blockchain_db/lmdb/db_lmdb.cpp]: Oh, right, for some reason I was thinking it was just a `uint64_t`.

**Pull request merged**